core/include/detray/propagator/navigation_policies.hpp

```diff
@@ -56,23 +56,26 @@ struct guided_navigation : actor {
 /// constraint was triggered.
 struct stepper_default_policy : actor {
 
-    struct state {};
+    struct state {
+        const scalar tol{std::numeric_limits<scalar>::epsilon()};
```

Comment:
Hm, we stay in the ```high trust``` case for a long time, which can lead to inconsistencies. I was worried, that might happen because of rounding errors or something. So I wanted to give it a tolerance saying, if you are very near the constraint, you are already in dangerous territory.